### PR TITLE
Add ESLint code style checking

### PR DESCRIPTION
**Pull request by am11**
* Add `.eslintrc.json` with rules matching the existing code.
  * Full list can be found at http://eslint.org/docs/rules/
* Reconcile the formatting in source and tests based on rules.
* Make CI to test formatting's.

### CODE REVIEW

**Comment by am11** [comment]
@nschonni, and this one is on top of changes in #97. :)

File: middleware.js [hunk 1]
```diff
@@ -128,7 +128,7 @@ module.exports = function(options) {
     }
 
     if (options.prefix) {
-      if (0 === path.indexOf(options.prefix)) {
+      if (path.indexOf(options.prefix) === 0) {
```
Comment (am11) [inline middleware.js]: Since `disallowYodaConditions` is true in `.jscsrc`.

File: test/scssTests.js [hunk 2]
```diff
@@ -7,29 +7,29 @@ var fs = require('fs'),
...
+    indexSourceMap = fixture('index.css.map'),
     testUtils = require('./testUtils');
 
 describe('Using middleware to compile .scss', function() {
   var server = connect()
     .use(middleware({
       src: fixture(),
-      dest: fixture()
+      dest: fixture(),
+      indentedSyntax: false
```
Comment (am11) [inline test/scssTests.js]: Added `indentedSyntax: false` in `sassTests.js` (although the default value of said option is `false`), to minimize the diff between `sassTests.js` and `scssTest.js` files.

**Comment by am11** [comment]
@xzyfer, should we add this kind of `jscs` infra in node-sass repo as well?

**Comment by nschonni** [comment]
node-sass switched to ESLint since JSCS merged with that project. Might make sense to do that here

**Comment by am11** [comment]
Ah didn't realized that, I will repurpose this PR for ESLint then. Thanks Nick!

**Comment by xzyfer** [comment]
@am11 I had a similar PR for jscs but @nschonni convinced me that ESLint was the better alternative for the reasons mentioned above.

File: middleware.js (hunk 1 above)
Comment (am11) [inline middleware.js]: With ESLint, the rule has simply become: `"yoda": 2`.

**Comment by am11** [comment]
Migrated to ESLint.
I have added few additional rules, however, for most part `.eslintrc.json` it's consistent with the one at `node-sass` root.

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,7 @@
 language: node_js
 node_js:
-  - '0.10'
-  - '0.12'
-  - '5'
+  - '6'
```
Comment (xzyfer) [inline .travis.yml]: I suggest LTS, and current which is currently 4, 6, 7. Node Sass will be shifting it's support this configuration later this year.

**Comment by xzyfer** [comment]
lgtm

File: .travis.yml (hunk 3 above)
Comment (am11) [inline .travis.yml]: Fixed by 3c455b8. Thanks! :)

**Pull request merged**